server/controllers/article.controllers.js

```diff
@@ -1,130 +1,139 @@
-        slug: article.slug,
-        abstract: article.abstract,
-        body: article.body,
-        category: article.category,
-        imageurl: article.image_url,
-        bookmarkcount: article.bookmark_count,
-        likescount: article.likes_count,
-        createdAt: article.createdAt,
-        updatedAt: article.updatedAt,
-      };
-
-      return res.status(200).json({
-        data: [articleObj],
-      });
-    } catch (error) {
-      return res.status(500).json({
-        error,
-        message: 'Oops! There seems to be a database error',
+    if (!article) {
+      return res.status(404).json({
+        errors: {
+          body: ['Article not found'],
+        },
       });
     }
-  },
-  async createArticle(req, res) {
-    try {
-      req.body.slug = slugMaker(req.body.title);
-      if (!req.body.slug) {
-        return res.status(400).json({
-          error: 'Article must have title',
-        });
-      }
-
-      const { userObj } = req.user;
-      req.body.user_id = userObj.id;
 
-      req.body = spaceTrimmer(req.body);
-      const article = await Article.create(req.body);
+    const articleObj = {
+      id: article.id,
+      author: article.user_id,
+      title: article.title,
+      slug: article.slug,
+      abstract: article.abstract,
+      body: article.body,
+      category: article.category,
+      imageurl: article.image_url,
+      bookmarkcount: article.bookmark_count,
+      likescount: article.likes_count,
+      createdAt: article.createdAt,
+      updatedAt: article.updatedAt,
+    };
 
-      if (!req.body.is_draft) {
-        if (req.body.keywords) {
-          req.body.keywords.forEach(async keyword => {
-            await tagsHelpers.saveArticleTags(article.id, keyword);
-          });
-        }
-      }
+    return res.status(200).json({
+      article: articleObj,
+    });
+  } catch (error) {
+    return res.status(500).json({
+      errors: serverError(),
+    });
+  }
+};
 
-      return res.status(201).json({
-        message: validations.draftPublishMessage(article.is_draft),
-        data: article,
-      });
-    } catch (error) {
-      return res.status(500).json({
-        error,
-        message: 'Oops! There seems to be a database error',
+const createArticle = async (req, res) => {
+  try {
+    req.body.slug = slugMaker(req.body.title);
+    if (!req.body.slug) {
+      return res.status(400).json({
+        errors: {
+          body: ['Article must have a title'],
+        },
       });
     }
-  },
-  async deleteArticle(req, res) {
-    try {
-      if (!validations.verifyUUID(req.params.id)) {
-        return res.status(400).json({
-          error: 'id not valid',
+
+    const { userObj } = req.user;
+    req.body.user_id = userObj.id;
+
+    req.body = spaceTrimmer(req.body);
+    const article = await Article.create(req.body);
+
+    if (!req.body.is_draft) {
+      if (req.body.keywords) {
+        req.body.keywords.forEach(async keyword => {
```

Comment:
I'm still seeing the double if I talked about @sojida weren't you able to fix this?

Reply:
@rotimi-babalola I intend to raise a PR to fix this when this code refactor PR is merged